cpp/src/arrow/record_batch_test.cc

```diff
@@ -393,6 +399,34 @@ TEST_F(TestRecordBatch, RemoveColumnEmpty) {
   AssertBatchesEqual(*added, *batch1);
 }
 
+TEST_F(TestRecordBatch, ColumnsThreadSafety) {
+  constexpr size_t kNumThreads = 10;
```

Comment:
Do we need to skip test when !ARROW_ENABLE_THREADING?

Reply:
Indeed the emscripten has [failed](https://github.com/ursacomputing/crossbow/actions/runs/13154843593/job/36709543256) with:
```
[ RUN      ] TestRecordBatch.ColumnsThreadSafety
unknown file: Failure
C++ exception with description "thread constructor failed: Resource temporarily unavailable" thrown in the test body.
[  FAILED  ] TestRecordBatch.ColumnsThreadSafety (2 ms)
```